src/components/manager/dropBox/DropBoxJsonSelect.tsx

```diff
@@ -0,0 +1,74 @@
+import { useEffect, useState } from "react";
+import { Radio } from "antd";
+
+import { BENTO_DROP_BOX_FS_BASE_PATH } from "@/config";
+import { useDropBoxJsonContent } from "@/modules/dropBox/hooks";
+import { dropBoxTreeNodeEnabledJson } from "@/utils/files";
+
+import DropBoxTreeSelect from "./DropBoxTreeSelect";
+import JsonDisplay from "@/components/display/JsonDisplay";
+import type { JSONType } from "@/types/json";
+
+const InnerDropBoxJsonSelect = ({
+  selectedFile,
+  setSelectedFile,
+}: {
+  selectedFile?: string;
+  setSelectedFile: (x: string) => void;
+}) => {
+  return (
+    <DropBoxTreeSelect
+      basePrefix={BENTO_DROP_BOX_FS_BASE_PATH}
+      multiple={false}
+      nodeEnabled={dropBoxTreeNodeEnabledJson}
+      allowClear={true}
+      value={selectedFile}
+      onChange={(v) => setSelectedFile(v)}
+    />
+  );
+};
+
+export type DropBoxJsonSelectProps = {
+  initialValue?: JSONType;
+  value?: JSONType;
+  onChange?: (x: JSONType) => void;
+};
+
+/** A form input component for DropBox JSON file selection. */
+const DropBoxJsonSelect = ({ initialValue, onChange }: DropBoxJsonSelectProps) => {
```

Comment:
Way better, thanks! I think I might add a radio option for "None", since Discovery config is optional.
The user can submit an empty discovery config by selecting nothing, but I don't think it is clear enough to the user.

Reply:
that's a good idea!